### PR TITLE
:sparkles: Expose emailConfirmedAt field from admin getRepo

### PR DESCRIPTION
**Pull request by foysalit**
This will help us view which users have verified their email address and make informed moderation decisions.

### CODE REVIEW

File: packages/pds/src/db/tables/repo-root.ts
Comment (devinivy) [inline packages/pds/src/db/tables/repo-root.ts:8]: I believe this column comes from the `user_account` table rather than the `repo_root` table.

File: packages/pds/tests/admin/get-repo.test.ts
Comment (devinivy) [inline packages/pds/tests/admin/get-repo.test.ts:113]: I think you can use `sc.getHeaders(sc.dids.bob)` here rather than manually grabbing Bob's token and building the header.

**Comment by foysalit** [comment]
Addressed both @devinivy thanks!

File: packages/pds/src/services/moderation/views.ts
Comment (devinivy) [inline packages/pds/src/services/moderation/views.ts:611]: Is there a reason this field should be optional?  My concern is that it could create inconsistencies in the resulting views, where some responses contain `emailConfirmedAt` and others don't.

File: packages/pds/src/services/moderation/views.ts
Comment (devinivy) [inline packages/pds/src/services/moderation/views.ts:24]: I think we can nix this now that it isn't used 👍

File: packages/pds/src/services/moderation/views.ts
Comment (foysalit) [reply packages/pds/src/services/moderation/views.ts:184]: so, the idea here is to only expose this field through `admin.getRepo()` endpoint and the lexicon property `com.atproto.admin.defs#repoViewDetail`. the primary use-case for this right now is to just show email verified accounts with a special flag on ozone's repository view page so trying to avoid exposing this on every `repoView` slice.
which is also why the field itself is defined as optional since the only time we're actually fetching it from the db before passing it along to the view is from the `account().getAccount()` [call](https://github.com/bluesky-social/atproto/blob/main/packages/pds/src/api/com/atproto/admin/getRepo.ts#L13)

LMK if that tracks.

File: packages/pds/src/services/moderation/views.ts
Comment (foysalit) [reply packages/pds/src/services/moderation/views.ts:611]: clarified in the [above comment](https://github.com/bluesky-social/atproto/pull/1757#discussion_r1376151206)

File: packages/pds/src/services/moderation/views.ts
Comment (devinivy) [reply packages/pds/src/services/moderation/views.ts:183]: Ahh, I am getting it now, thanks!

Since `emailConfirmedAt` is part of `RepoViewDetail` but not `RepoView`, I think it would be sort of nice/tidy for that difference to be reflected in `repoDetail()` the same way the other differences between those two views are.  With this change there's a bit of coupling between how the result set is generated versus whether it's suitable to be passed to `repo()` versus `repoDetail()`, and I wonder if it might be a little simpler to always select `emailConfirmedAt` then place the logic to include the field in the view inside `repoDetail()`.

All that said this is quite a small thing, I certainly wouldn't block on this 👍

**Review by devinivy — COMMENTED** [review]
Looks good!  However, I think post-merge you'll find that many of the existing PDS changes can go away and that `admin.getAccountInfo` will need to be updated.  The appview's `admin.getRepo` looks like it now calls out to the PDS's `admin.getAccountInfo` for invite/email details.

**Comment by foysalit** [comment]
@devinivy brought in some more changes, mind giving it one more look before I hit merge?

**Review by devinivy — APPROVED** [review]
Great!

**Pull request merged**
